pkg/infrastructure/task.go

```diff
@@ -0,0 +1,84 @@
+package infrastructure
+
+import (
+	"context"
+	"time"
+
+	"github.com/ari1021/hack-ios-server/hconstant"
+	"github.com/ari1021/hack-ios-server/pkg/domain/entity"
+	"github.com/ari1021/hack-ios-server/pkg/domain/repository"
+	"gorm.io/gorm"
+)
+
+type Task struct {
+	ID          string    `gorm:"primaryKey"`
+	CreatedAt   time.Time `gorm:"index"`
+	UpdatedAt   time.Time
+	DeletedAt   gorm.DeletedAt `gorm:"index"`
```

Comment:
どちらでも同じことになると思うけど、ここだけgorm.Deletedになってるので

```suggestion
	DeletedAt   *time.Time `gorm:"index"`
```

にしといたほうが良さそう？

Reply:
`gorm.DeletedAt`は，gormを扱った際のFindのときなどに自動で論理削除されたデータを省いてくれるので，`gorm.DeletedAt`と`*time.Time`は微妙に違う気がするのですがどうでしょうか？

Reply:
どちらかに統一はした方が良くて、gormの方が使い勝手はいいかも(前のレビューではtimeのポインタを示したけど

Reply:
`entity`内でgorm.DeletedAtを使うのが気持ち悪い(entityに技術的な部分を持ち込みたくない)のですが，どう思いますか？

Reply:
あーそうか、それならポインタでよさそう

Reply:
統一されてなくてもok
混乱させてごめん！

Reply:
> gorm.DeletedAtは，gormを扱った際のFindのときなどに自動で論理削除されたデータを省いてくれる

なるほど！そういう理由か！
であれば `gorm.DeletedAt` を使うでよさそう 👍

Reply:
> entity内でgorm.DeletedAtを使うのが気持ち悪い

entity では `*time.Time` を使うのままで！